### PR TITLE
 Minimum in Binary Search Tree(BST) Documentation

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#333 )

### CODE REVIEW

File: docs/data-structures/minimum-in-bst.mdx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+```
+Minimum element in BST: 11
+```
+
+## Complexity Analysis
+ 
+- **Time Complexity:** `O(h)`
+
+- **Space Complexity:** `O(1)`
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:191]: Don't keep Time & Space Complexity in bold

File: docs/data-structures/minimum-in-bst.mdx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+where `n` is the number of elements and `h` is the height of binary tree.
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:193]: Replace `where` with `here`

File: docs/data-structures/minimum-in-bst.mdx [hunk 3]
```diff
@@ -0,0 +1,193 @@
+---
+title: Minimum in Binary Search Tree(BST)
+slug: /data-structures/minimum-in-bst
+---
+
+## Introduction
+
+In this article, we will see how to find the node having minimum data or key. We are given an Binary Search Tree(BST) and we have to find the minimum element. In this type of tree(BST), left child is always smaller than its parent and right child is always greater than its parent.
+
+## Explanation
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:10]: Keep a blank line after headings

File: docs/data-structures/minimum-in-bst.mdx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+For finding minimum node in BST, we follow the following algorithm:
+
+* First, we will check that tree is empty or not.
+
+* If tree is empty then simply return 0 because it has no node.
+
+* Otherwise keep moving on left child until we get left child as NULL.
+
+* Print the left child because this is the minimum node in BST.
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:34]: Don't keep any blank lines here.

File: docs/data-structures/minimum-in-bst.mdx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+## Introduction
+
+In this article, we will see how to find the node having minimum data or key. We are given an Binary Search Tree(BST) and we have to find the minimum element. In this type of tree(BST), left child is always smaller than its parent and right child is always greater than its parent.
+
+## Explanation 
+In this we are given an Binary Search Tree and we have to find out the node which have minimum data or key. We know that in this type of tree left child is smaller than its parent and right child is always greater than its parent. So the minimum data will lie in left part of tree so keep moving in left until we get null.
+#### Example 1
+<img src="https://i.imgur.com/vjoI6QE.jpg?1" width="600" height="350"/>
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:13]: Did you make this image?

File: docs/data-structures/minimum-in-bst.mdx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+slug: /data-structures/minimum-in-bst
+---
+
+## Introduction
+
+In this article, we will see how to find the node having minimum data or key. We are given an Binary Search Tree(BST) and we have to find the minimum element. In this type of tree(BST), left child is always smaller than its parent and right child is always greater than its parent.
+
+## Explanation 
+In this we are given an Binary Search Tree and we have to find out the node which have minimum data or key. We know that in this type of tree left child is smaller than its parent and right child is always greater than its parent. So the minimum data will lie in left part of tree so keep moving in left until we get null.
+#### Example 1
```
Comment (HarshCasper) [inline docs/data-structures/minimum-in-bst.mdx:12]: Rename to `Examples`. Keep a blank line before and after the heading

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/minimum-in-bst.mdx (hunk 5 above)
Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx:13]: Yes

Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx:13]: Yes

**Comment by shivammahajancse** [comment]
Changes done sir @HarshCasper @iamrajiv @atarax665

**Comment by shivammahajancse** [comment]
please review @ankitaggarwal23 sir

**Review by iamrajiv — APPROVED** [review]

File: docs/data-structures/minimum-in-bst.mdx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include<iostream>
```
Comment (ankitaggarwal23) [inline docs/data-structures/minimum-in-bst.mdx:51]: Replace header with `#include <bits/stdc++.h>`

File: docs/data-structures/minimum-in-bst.mdx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+//Main Function began
+int main() {
+    Node* root = NULL;
+    //input values
+     int n;
+     cout<<"Enter the size of tree: ";
+     cin>>n;
+    while(n--){
```
Comment (ankitaggarwal23) [inline docs/data-structures/minimum-in-bst.mdx:97]: Align you code properly.

File: docs/data-structures/minimum-in-bst.mdx [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+        int d;
+        cout<<"Enter the data: ";
+        cin>>d;
+        root = Insert(root, d);
+    }
+
+    cout << "Minimum element in BST: "<<minimum_ele(root);
+    
+    cout << "\n";
```
Comment (ankitaggarwal23) [inline docs/data-structures/minimum-in-bst.mdx:106]: Rewrite this line as,
`cout << "Minimum element in BST: "<<minimum_ele(root) << endl;`

File: docs/data-structures/minimum-in-bst.mdx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+{
+  struct Node* curr=root;
+  while(curr->left!=NULL)
+  {
+    curr=curr->left;
+  }
+  return curr->data;
+}
+
+//Main Function began
```
Comment (ankitaggarwal23) [inline docs/data-structures/minimum-in-bst.mdx:90]: Replace this redundant comments as,

`// Driver Code`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Kindly use python linters and use proper spacing in python code. Too much space in python code.

File: docs/data-structures/minimum-in-bst.mdx (hunk 7 above)
Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx]: done

File: docs/data-structures/minimum-in-bst.mdx (hunk 8 above)
Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx]: done

File: docs/data-structures/minimum-in-bst.mdx (hunk 9 above)
Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx]: done

File: docs/data-structures/minimum-in-bst.mdx (hunk 10 above)
Comment (shivammahajancse) [reply docs/data-structures/minimum-in-bst.mdx]: done

**Comment by shivammahajancse** [comment]
@ankitaggarwal23  sir all changes done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**